src/msg/simple/Pipe.cc

```diff
@@ -901,6 +901,7 @@ int Pipe::connect()
   // create socket?
   sd = ::socket(peer_addr.get_family(), SOCK_STREAM, 0);
   if (sd < 0) {
+    rc = -errno;
     lderr(msgr->cct) << "connect couldn't created socket " << cpp_strerror(errno) << dendl;
```

Comment:
Thanks Josh for your inputs, sorry I missed to explain this to Kefu, why we did like this.